File: antd/src/services/dnf-admin/daFridaFunctionController.ts
// @ts-ignore
/* eslint-disable */
import { request } from 'umi';

/** 获取引入的函数列表 GET /api/fridaFunction/getChildrenFunction/${param0} */
export async function getChildrenFunction(
  // 叠加生成的Param类型 (非body参数swagger默认没有生成对象)
  params: API.getChildrenFunctionParams,
  options?: { [key: string]: any },
) {
  const { id: param0, ...queryParams } = params;
  return request<API.RListDaFridaFunction>(`/api/fridaFunction/getChildrenFunction/${param0}`, {
    method: 'GET',
    params: { ...queryParams },
    ...(options || {}),
  });
}

/** 根据frida函数信息主键获取详细信息 GET /api/fridaFunction/getInfo/${param0} */
export async function getFridaFunctionInfo(
  // 叠加生成的Param类型 (非body参数swagger默认没有生成对象)
  params: API.getFridaFunctionInfoParams,
  options?: { [key: string]: any },
) {
  const { id: param0, ...queryParams } = params;
  return request<API.RDaFridaFunction>(`/api/fridaFunction/getInfo/${param0}`, {
    method: 'GET',
    params: { ...queryParams },
    ...(options || {}),
  });
}

/** 查询所有frida函数信息 GET /api/fridaFunction/list */
export async function listFridaFunction(options?: { [key: string]: any }) {
  return request<API.RListDaFridaFunction>('/api/fridaFunction/list', {
    method: 'GET',
    ...(options || {}),
  });
}

/** 分页查询frida函数信息 POST /api/fridaFunction/page */
export async function pageFridaFunction(
  body: API.FridaFunctionQo,
  options?: { [key: string]: any },
) {
  return request<API.RListDaFridaFunction>('/api/fridaFunction/page', {
    method: 'POST',
    headers: {
      'Content-Type': 'application/json',
    },
    data: body,
    ...(options || {}),
  });
}

/** 添加frida函数信息 POST /api/fridaFunction/save */
export async function saveFridaFunction(
  body: API.DaFridaFunction,
  options?: { [key: string]: any },
) {
  return request<API.RBoolean>('/api/fridaFunction/save', {
    method: 'POST',
    headers: {
      'Content-Type': 'application/json',
    },
    data: body,
    ...(options || {}),
  });
}

/** 根据主键更新frida函数信息 POST /api/fridaFunction/update */
export async function updateFridaFunction(
  body: API.DaFridaFunction,
  options?: { [key: string]: any },
) {
  return request<API.RBoolean>('/api/fridaFunction/update', {
    method: 'POST',
    headers: {
      'Content-Type': 'application/json',
    },
    data: body,
    ...(options || {}),
  });
}

File: be/src/main/java/plus/easydo/dnf/controller/DaFridaFunctionController.java
package plus.easydo.dnf.controller;

import cn.dev33.satoken.annotation.SaCheckLogin;
import cn.dev33.satoken.annotation.SaCheckPermission;
import lombok.RequiredArgsConstructor;
import org.springframework.web.bind.annotation.GetMapping;
import org.springframework.web.bind.annotation.PathVariable;
import org.springframework.web.bind.annotation.PostMapping;
import org.springframework.web.bind.annotation.RequestBody;
import org.springframework.web.bind.annotation.RequestMapping;
import org.springframework.web.bind.annotation.RestController;
import plus.easydo.dnf.entity.DaFridaFunction;
import plus.easydo.dnf.qo.FridaFunctionQo;
import plus.easydo.dnf.service.IDaFridaFunctionService;
import plus.easydo.dnf.vo.DataResult;
import plus.easydo.dnf.vo.R;

import java.io.Serializable;
import java.util.List;

/**
 * frida函数信息 控制层。
 *
 * @author mybatis-flex-helper automatic generation
 * @since 1.0
 */
@RestController
@RequiredArgsConstructor
@RequestMapping("/api/fridaFunction")
public class DaFridaFunctionController {

    private final IDaFridaFunctionService daFridaFunctionService;

    /**
     * 添加 frida函数信息
     *
     * @param daFridaFunction frida函数信息
     * @return {@code true} 添加成功，{@code false} 添加失败
     */
    @SaCheckPermission("function.save")
    @PostMapping("/save")
    public R<Boolean> saveFridaFunction(@RequestBody DaFridaFunction daFridaFunction) {
        return DataResult.ok(daFridaFunctionService.saveFridaFunction(daFridaFunction));
    }



    /**
     * 根据主键更新frida函数信息
     *
     * @param daFridaFunction frida函数信息
     * @return {@code true} 更新成功，{@code false} 更新失败
     */
    @SaCheckPermission("function.update")
    @PostMapping("/update")
    public R<Boolean> updateFridaFunction(@RequestBody DaFridaFunction daFridaFunction) {
        return DataResult.ok(daFridaFunctionService.updateFridaFunction(daFridaFunction));
    }


    /**
     * 查询所有frida函数信息
     *
     * @return 所有数据
     */
    @SaCheckLogin
    @GetMapping("/list")
    public R<List<DaFridaFunction>> listFridaFunction() {
        return DataResult.ok(daFridaFunctionService.listFridaFunction());
    }


    /**
     * 根据frida函数信息主键获取详细信息。
     *
     * @param id daFridaFunction主键
     * @return frida函数信息详情
     */
    @SaCheckPermission("function")
    @GetMapping("/getInfo/{id}")
    public R<DaFridaFunction> getFridaFunctionInfo(@PathVariable Long id) {
        return DataResult.ok(daFridaFunctionService.getInfo(id));
    }

    /**
     * 根据frida函数信息主键获取详细信息。
     *
     * @param id daFridaFunction主键
     * @return frida函数信息详情
     */
    @SaCheckPermission("function")
    @GetMapping("/getChildrenFunction/{id}")
    public R<List<DaFridaFunction>> getChildrenFunction(@PathVariable Long id) {
        return DataResult.ok(daFridaFunctionService.getChildrenFunction(id));
    }


    /**
     * 分页查询frida函数信息
     *
     * @param fridaFunctionQo 分页对象
     * @return 分页对象
     */
    @SaCheckPermission("function")
    @PostMapping("/page")
    public R<List<DaFridaFunction>> pageFridaFunction(@RequestBody FridaFunctionQo fridaFunctionQo) {
        return DataResult.ok(daFridaFunctionService.pageFridaFunction(fridaFunctionQo));
    }
}
